rust/lance-io/src/object_store/gcs_wrapper.rs

```diff
@@ -330,7 +375,7 @@ impl AsyncWrite for Upload {
                     ..
                 } => {
                     // TODO: Make max concurrency configurable.
-                    if futures.len() < MAX_UPLOAD_PARALLELISM {
+                    if futures.len() < max_upload_parallelism() {
```

Comment:
We can still exceed this if we have multiple writers right?

Reply:
Yeah. I thought about trying to integrate limit store, but:
1. In `object-store` 0.9.x it doesn't handle multipart uploads well. 0.10.x does this better.
2. Because we don't have a central store, each call to `write_fragments` or `write_dataset` creates it's own object store instance with it's own limits
3. For users of Ray and similar frameworks, their stores are in entirely different processes, so a limit store would have no effect.

So I felt at least we can try letting users manually tune through these variables.